packages/package/src/index.js

```diff
@@ -22,25 +22,29 @@ export async function build(options) {
  * @param {(name: string, code: string) => void} analyse_code
  */
 async function do_build(options, analyse_code) {
-	const { input, output, extensions, alias } = normalize_options(options);
+	const { input, output, temp, extensions, alias } = normalize_options(options);
 
 	if (!fs.existsSync(input)) {
 		throw new Error(`${path.relative('.', input)} does not exist`);
 	}
 
-	rimraf(output);
-	mkdirp(output);
+	rimraf(temp);
+	mkdirp(temp);
 
 	const files = scan(input, extensions);
 
 	if (options.types) {
-		await emit_dts(input, output, options.cwd, alias, files);
+		await emit_dts(input, temp, options.cwd, alias, files);
 	}
 
 	for (const file of files) {
-		await process_file(input, output, file, options.config.preprocess, alias, analyse_code);
+		await process_file(input, temp, file, options.config.preprocess, alias, analyse_code);
 	}
 
+	rimraf(output);
```

Comment:
Does this delete the directory or empty it? If delete then is it preferable to empty it, or would that not help with the issue?

Reply:
It deletes it...like `rm -rf`. I imagined I needed something more involved when I started but `rimraf` and `copy` were just there and they worked. Because the rimraf and copy happen in quick succession it seems to be good enough. I tested this approach when I created the repro for the issue, and I've also used it in another repo a few times today too.

Reply:
Ah awesome!